docs/topics/gradle/gradle-compilation-and-caches.md

```diff
@@ -283,6 +284,57 @@ tasks.withType(KotlinCompile)
 </tab>
 </tabs>
 
+## Kotlin compiler fallback strategy
+
+The Kotlin compiler's fallback strategy is to run a compilation outside a Kotlin daemon if the daemon somehow fails. 
+If the Gradle daemon is on, the compiler uses the ["In process" strategy](#defining-kotlin-compiler-execution-strategy). 
+If the Gradle daemon is off, the compiler uses the "Out of process" strategy.
+
+When this fallback happens, you have the following warning lines in your Gradle's build output:
+
+```
+Failed to compile with Kotlin daemon: java.lang.RuntimeException: Could not connect to Kotlin compile daemon
+[exception stacktrace]
+Using fallback strategy: Compile without Kotlin daemon
+Try ./gradlew --stop if this issue persists.
+```
+
+However, a silent fallback to another strategy can consume a lot of system resources or lead to non-deterministic builds, 
+read more about this in this [YouTrack issue](https://youtrack.jetbrains.com/issue/KT-48843/Add-ability-to-disable-Kotlin-daemon-fallback-strategy).
+To avoid this, there is a Gradle property `kotlin.daemon.useFallbackStrategy`, whose default value is `true`. 
```

Comment:
@sarahhaggarty Could you please help me with this sentence? Does it sound weird with this beginning?

Reply:
Thanks!